indico/modules/events/abstracts/legacy.py

```diff
@@ -0,0 +1,498 @@
+            fval = self.event.abstract_fields.filter_by(contribution_field=field, abstract=self.as_new).first()
+            if fval:
+                return fval
+            else:
+                return AbstractFieldValue(contribution_field=field, abstract=self.as_new, data={})
+
+    @property
+    @memoize_request
+    def event(self):
+        return self._owner._owner.as_event
+
+    @property
+    @memoize_request
+    def as_new(self):
+        return Abstract.find_one(event_new=self.event, legacy_id=self._id)
+
+    @no_autoflush
+    def _add_judgment(self, legacy_judgment):
+        judgment = Judgment(track_id=legacy_judgment._track.id, judge=legacy_judgment._responsible.user,
+                            accepted_type=getattr(legacy_judgment, '_contribType', None))
+        self.as_new.judgments.append(judgment)
+        return judgment
+
+    def _del_judgment(self, legacy_judgment):
+        db.session.delete(legacy_judgment.as_new)
+        db.session.flush()
+
+    def getFields(self):
+        data = {val.contribution_field.legacy_id: AbstractFieldContentWrapper(val)
+                for val in self.as_new.field_values}
+        data['content'] = self.as_new.description
+        return data
+
+    def getField(self, field_id):
+        field_val = self._get_field_value(field_id)
+        return AbstractFieldContentWrapper(field_val)
+
+    def removeField(self, field_id):
+        field_val = self._get_field_value(field_id)
+        if field_val:
+            self.as_new.field_values.remove(field_val)
+
+    def setField(self, field_id, val):
+        field_val = self._get_field_value(field_id)
+        field_val.data = val
+
+    def getContribType(self):
+        return self.as_new.type
+
+
+class AbstractFieldContentWrapper(object):
+    """Emulates legacy ``AbstractFieldContent`` object."""
+
+    def __init__(self, field_val):
+        self.field_val = field_val
+
+    @property
+    def field(self):
+        return self.field_val.contribution_field
+
+    @property
+    def value(self):
+        return self.field_val.data
+
+    def __eq__(self, other):
+        if isinstance(other, AbstractFieldContentWrapper) and self.field.id == other.field.id:
+            return self.field_val.data == other.data
+        elif not isinstance(other, AbstractFieldContentWrapper):
+            return self.field_val.data == other
+        return False
+
+    def __len__(self):
+        return len(self.field_val.data)
+
+    def __ne__(self, other):
+        if isinstance(other, AbstractFieldContentWrapper) and self.field.id == other.field.id:
+            return self.field_val.data != other.data
+        elif not isinstance(other, AbstractFieldContentWrapper):
+            return self.field_val.data != other
+        return True
+
+    @encode_utf8
+    def __str__(self):
+        if self.field.field_type == 'single_choice':
+            return unicode(AbstractSelectionFieldWrapper(self.field).getOption(self.value))
```

Comment:
`unicode(...)` works only when we have only the `__str__` method. I guess that's because we're not explicitly encoding anything. Anyway, I'll add `__unicode__`.